### DISCUSSION TITLE
weed proxy without state

### DISCUSSION THREAD

**Discussion by bingoohuang** [seaweedfs/seaweedfs#1826] [Ideas]
I want a filer-alike component ( here i temporary name it weed-proxy), which help me do the following tasks:

1. encrypt/decrypt for the data in customized algrithm
2. customized fid assigning(eg. assign 5 fids which spread among volume servers as much as possible)
3. multiple little files uploading allowed
4. customized authc/authz on use, group, volume, collection, fids or other properties
5. and etc.

 the great seaweedfs author @chrislusf,  any any advice from you?

**Comment by chrislusf** [comment]
You can build anything to replace filer. Actually filer was initially developed just to provide a referential implementation of how to use the blob storage.

There could be many more data systems built on top of blob storage, without the "file system". E.g.

- A Kafka like message queue but storing data to blob storage
- A big data storage and querying engine
- A time-series database
- A full text search system
